### PR TITLE
P4-1585 add booked move status

### PR DESCRIPTION
**Pull request by martyn-w**
### Jira link

[P4-1585](https://dsdmoj.atlassian.net/browse/P4-1585)

### What?

I have added/removed/altered:

- [x] Adds new booked move status

### Why?

I am doing this because:

- because this status forms part of supplier assurance testing

### Have you? (optional)

- [x] Updated API docs if necessary	

### Deployment risks (optional)

- high risk: the FE will not be expecting this status

### CODE REVIEW

File: spec/models/move_spec.rb [hunk 1]
```diff
@@ -61,6 +61,12 @@
     )
   end
 
+  it 'does NOT validates presence of `profile` if `status` is booked' do
```
Comment (willfish) [inline spec/models/move_spec.rb:64]:
```suggestion
  it 'does NOT validate presence of `profile` if `status` is booked' do
```

File: spec/requests/api/v1/moves_controller_index_spec.rb [hunk 2]
```diff
@@ -91,6 +91,25 @@
         # rubocop:enable RSpec/ExampleLength
       end
 
+      context 'with a booked move' do
```
Comment (willfish) [inline spec/requests/api/v1/moves_controller_index_spec.rb:94]: AFAIK contexts are used to describe specific contexts that affect the behaviour of the code. My sense is that this test and context doesn't add any value as far as testing behaviour is concerned? I could be wrong, though.

Comment (willfish) [reply spec/requests/api/v1/moves_controller_index_spec.rb:94]: I mean, nothing really varies that could change the behaviour of how we index?

Comment (willfish) [reply spec/requests/api/v1/moves_controller_index_spec.rb:94]: I think you're capturing this in the finder, spec, too :).

File: lib/tasks/fake_data.rake [hunk 3]
```diff
@@ -190,7 +190,7 @@ namespace :fake_data do
           profile: profile,
           from_location: from_location,
           to_location: to_location,
-          status: %w[proposed requested completed].sample,
+          status: %w[proposed requested booked completed].sample,
```
Comment (willfish) [inline lib/tasks/fake_data.rake:193]: 🎉

**Comment by willfish** [comment]
So this isn't dangerous to the frontend currently, right? Because we don't expose the ability to mark a move's status as booked?

**Comment by martyn-w** [comment]
@willfish its possible a move status could be set to 'booked' via the API, then the key question is, how will the FE behave ?

**Comment by willfish** [comment]
Ah, then probably running the e2e tests against this makes sense

**Comment by willfish** [comment]
In conjunction with creating a booked move

**Comment by martyn-w** [comment]
The FE will need updating to show booked moves; they are currently invisible if the status=booked

# requested move
Screenshot 2020-05-29 at 17 09 39

# booked move
Screenshot 2020-05-29 at 17 09 55

**Comment by martyn-w** [comment]
Closing for now, will re-open when this work can be unblocked.

**Comment by martyn-w** [comment]
Re-opening to allow suppliers to change a move's status to be `booked`, on the understanding that booked moves will not be visible until the front end is updated. This is necessary for the next round of API assurance. cc @jason-sims

File: app/models/move.rb [hunk 4]
```diff
@@ -53,7 +55,7 @@ class Move < VersionedModel
   validates :from_location, presence: true
   validates :to_location, presence: true, unless: :prison_recall?
   validates :move_type, inclusion: { in: move_types }
-  validates :profile, presence: true, unless: -> { requested? || cancelled? }
+  validates :profile, presence: true, unless: -> { requested? || booked? || cancelled? }
```
Comment (smoothcontract) [inline app/models/move.rb:58]: Are you sure about this change? i.e. can a move can be booked (confirmed by the supplier) without a person specified?

Comment (martyn-w) [reply app/models/move.rb:58]: @smoothcontract hmm, that doesn't look correct - I'll fix it

**Pull request merged**